lib/ExpensiMark.js

```diff
@@ -261,7 +264,9 @@ export default class ExpensiMark {
                 name: 'quote',
                 regex: /\n?<(blockquote|q)(?:"[^"]*"|'[^']*'|[^'">])*>([\s\S]*?)<\/\1>(?![^<]*(<\/pre>|<\/code>))/gi,
                 replacement: (match, g1, g2) => {
-                    const resultString = g2.trim().split('\n').map(m => `> ${m}`).join('\n');
+                    const resultString = g2.replace(/\[block\]/g, '\n').trim()
+                        .split('\n').map(m => `> ${m}`)
+                        .join('\n');
```

Comment:
Why are we doing this here?

Reply:
@eh2077 bump!

Reply:
@Santhosh-Sellavel Thanks for looking into this.

As we translated `<h1>` tag into `[block]# $2[block]` using the separator `[block]` to handle line break between markdown block elements, like heading, code fence and quote, see
https://github.com/Expensify/expensify-common/blob/3cdaa947fe77016206c15e523017cd50678f2359/lib/ExpensiMark.js#L233-L237

Currently, we handle line break between markdown block element in method [replaceBlockWithNewLine](https://github.com/Expensify/expensify-common/blob/3cdaa947fe77016206c15e523017cd50678f2359/lib/ExpensiMark.js#L475-L501)
https://github.com/Expensify/expensify-common/blob/3cdaa947fe77016206c15e523017cd50678f2359/lib/ExpensiMark.js#L475-L501
While inside a quote, I think it'll be better to handle the `[block]` separators which wrap the heading here in the replacement method of html-blockquote-to-markdown-quote rule because
1. We'll only have one kind of markdown block element, heading, inside a quote
2. It's not that easy to modify method [replaceBlockWithNewLine](https://github.com/Expensify/expensify-common/blob/3cdaa947fe77016206c15e523017cd50678f2359/lib/ExpensiMark.js#L475-L501) to handle heading inside a quote

Reply:
When writing the above the comment, I found another case of having two continuous headers, like
```
===== normal heading =====
# heading A
# heading B
===== heading in quote =====
> # heading A inside quote
> # heading B inside quote
```
I fixed it and added tests. See below video

https://user-images.githubusercontent.com/117511920/234743733-d01b1a95-947c-45b3-8808-3def875fce29.mov